src/creature.cpp

```diff
@@ -168,16 +168,22 @@ bool Creature::hide( const tripoint &target, bool move )
 
 bool Creature::unhide( const tripoint entered_from )
 {
-    if( has_effect( effect_no_sight ) && !g->m.has_flag_ter_or_furn( TFLAG_NO_SIGHT, entered_from ) ) {
+    if( has_effect( effect_no_sight ) ) {
```

Comment:
With no other conditions this check is redundant. The same with `effect_hidden` below.

Reply:
Done